builder/dockerfile/parser/parser.go

```diff
@@ -101,19 +103,24 @@ func parseLine(line string) (string, *Node, error) {
 // Parse is the main parse routine.
 // It handles an io.ReadWriteCloser and returns the root of the AST.
 func Parse(rwc io.Reader) (*Node, error) {
+	currentLine := 0
 	root := &Node{}
+	root.StartLine = currentLine
```

Comment:
Hopefully this won't get into a deep bike shedding discussion :-) but the code needs to decide whether the line numbering is 0 or 1 based. I was assuming 1-based since you increment the currentLine value after each scanner.Text() call.  Which is good, I think that's probably correct from a UX perspective.

However, I think root.StartLine should be 1 not 0 then.  The above will set it to 0.  Perhaps remove the above line and down below when you increment currentLine, set StartLine to currentLine if its still 0.  Then StartLine will be 5 if there are 4 blank lines at the start of the file.  Likewise, it will still be 0 if the file has no real content - which might be useful info too.